tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/Keramelizer.scala

```diff
@@ -143,19 +143,47 @@ class Keramelizer(gen: UniqueNameGenerator, tracker: TransformationTracker)
    * @return a transformed expression
    */
   private def transformControl: PartialFunction[TlaEx, TlaEx] = {
+
     case expr @ OperEx(TlaControlOper.caseWithOther, otherEx, args @ _*) =>
-      def decorateWithIf(elseEx: TlaEx, guardAction: (TlaEx, TlaEx)): TlaEx = {
-        tla.ite(guardAction._1, guardAction._2, elseEx).typed(elseEx.typeTag.asTlaType1())
+      // CASE with a default arm
+      if (expr.typeTag == Typed(BoolT1())) {
+        // the Boolean case becomes a disjunction that has otherEx as an option
+        val (guards, actions) = TlaOper.deinterleave(args)
+        // produce g_1 /\ a_1, ..., g_n /\ a_n
+        val ands = guards.zip(actions) map { case (g, a) => tla.and(g, a).typed(BoolT1()) }
```

Comment:
Instead of `deinterleave` + `zip`, why not `args.grouped(2)`?

Reply:
Since we have to combine guards for the default case, no use of grouped anymore :)

Reply:
Fair enough